avocado/utils/cpu.py

```diff
@@ -188,3 +189,56 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpu_idle():
+    """
+    Get current cpu idle values
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idle_state = [[0]*len(states)]*len(cpus_list)
+    for c in cpus_list:
+        for state in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+            try:
+                cpu_idle_state[c][state] = int(open(state_file).read())
+            except Exception:
+                pass  # Ignore if states are not available
+    return cpu_idle_state
+
+
+def set_cpu_idle(state="all", disable=True, old_value=None):
```

Comment:
how about `set_cpuidle_state`? Then the `disable` can be turned to `value` and it'll become quite readable:

* set_cpuidle_state 1 to value 0
* set_cpuidle_state old_value=my_state